core/domain/email_manager_test.py

```
@@ -9210,3 +9210,24 @@ def mock_enqueue_task(
             enqueued_tasks[0][0], feconf.TASK_URL_RETRY_FAILED_EMAIL
         )
         self.assertEqual(enqueued_tasks[0][1]['subject'], 'Subject')
+
+    def test_failed_send_mail_does_not_create_sent_email_model(self) -> None:
+        def mock_send_mail(*_args: str, **_kwargs: str) -> None:
+            raise Exception('Simulated email failure')
```

Ruff (0.16.1)

[warning] Create your own exception

(TRY002)